daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_functions/Functions.tdml

```diff
@@ -459,6 +459,37 @@
       </xs:complexType>
     </xs:element>
 
+    <xs:element name="valueLength14">
+      <xs:complexType>
+        <xs:sequence dfdl:separator="%WSP;">
+          <xs:element name="first" type="xs:string" dfdl:lengthKind="delimited"/>
+          <xs:element name="other" type="xs:string" dfdl:lengthKind="delimited" maxOccurs="unbounded"/>
+          <xs:element name="len" type="xs:unsignedInt" dfdl:inputValueCalc='{dfdl:valueLength(../ex:other, "bytes")}'/>
+        </xs:sequence>
+      </xs:complexType>
+    </xs:element>
+
+    <xs:element name="valueLength15" type="xs:unsignedInt" dfdl:inputValueCalc='{dfdl:valueLength("three", "bytes")}'/>
+
+    <xs:element name="valueLength16">
+      <xs:complexType>
+        <xs:sequence>
+          <xs:element name="e1">
```

Comment:
We might want to make e1 an array (e.g. `maxOccurs="unbounded"`) since this issue was all about getting the length of arrays, ande make sure a self expression in a complex really is referencing the right thing. Same with the contentLength6 schema.

Reply:
Yeah, unbounded makes for a better test. I've added that change.

Reply:
You have made v1 an array, bu the valueLength path references e1 which is still a scalar. Can the maxOccurs unbounded be moved to the e1 element so it's a complex array?

Reply:
It has been moved up to the e1 element.